autorest/preparer_test.go

```diff
@@ -437,11 +439,82 @@ func TestWithFormDataSetsContentLength(t *testing.T) {
 		t.Fatalf("autorest: WithFormData failed with error (%v)", err)
 	}
 
+	expected := "name=Rob+Pike&age=42"
+	if sortString(expected) != sortString(string(b)) {
+		t.Fatalf("autorest:WithFormData failed to return correct string got (%v), expected (%v)", string(b), expected)
+	}
+
 	if r.ContentLength != int64(len(b)) {
 		t.Fatalf("autorest:WithFormData set Content-Length to %v, expected %v", r.ContentLength, len(b))
 	}
 }
 
+func sortString(x string) string {
+	s := strings.Split(x, "")
```

Comment:
shouldn't you split by &? for Rob Pike example above I'd just do if

    a == "name=Rob+Pike&age=42" || a == "age=42&name=Rob+Pike"

and that'd solve the problem